daml-foundations/daml-ghc/daml-prim-src/GHC/Enum.daml

```diff
@@ -69,61 +66,59 @@ class  Bounded a  where
 -- ```
 --
 class  Enum a   where
-    -- | the successor of a value.  For numeric types, 'succ' adds 1.
+    -- | Returns the successor of the given value. For example, for
+    -- numeric types, `succ` adds 1.
+    --
+    -- If the type is also an instance of `Bounded`, `succ maxBound`
+    -- results in a runtime error.
     succ                : a -> a
-    -- | the predecessor of a value.  For numeric types, 'pred' subtracts 1.
+    -- | Returns the predecessor of the given value. For example, for
+    -- numeric types, `pred` subtracts 1.
+    --
+    -- If the type is also an instance of `Bounded`, `pred minBound`
+    -- results in a runtime error.
     pred                : a -> a
-    -- | Convert from an 'Int'.
+    -- | Convert a value from an `Int` to an `Enum` value: ie, returns
+    -- the item at the `Int` position from the `Enum`.
+
     toEnum              : Int -> a
-    -- | Convert to an 'Int'.
-    -- It is implementation-dependent what 'fromEnum' returns when
-    -- applied to a value that is too large to fit in an 'Int'.
+    -- | Convert a value from an `Enum` value to an `Int`: ie, returns
+    -- the `Int` position of the element within the `Enum`.
+    --
+    -- If `fromEnum` is applied to a value that's too large to
+    -- fit in an `Int`, what is returned is up to your implementation.
     fromEnum            : a -> Int
 
-    -- | Used in Haskell's translation of `[n..]` with `[n..] = enumFrom n`,
-    --   a possible implementation being `enumFrom n = n : enumFrom (succ n)`.
-    --   For example:
+    -- | Return a list of the `Enum` values starting at the `Int`
+    -- position. For example:
     --
-    --     * `enumFrom 4 :: [Integer] = [4,5,6,7,...]`
-    --     * `enumFrom 6 :: [Int] = [6,7,8,9,...,maxBound :: Int]`
+    --     * `enumFrom 6 : [Int] = [6,7,8,9,...,maxBound : Int]`
     enumFrom            : a -> [a]
-    -- | Used in Haskell's translation of `[n,n'..]`
```

Comment:
I think this information is important. Same below for `enumFromTo`.

Reply:
Maybe it's just the phrasing I object to. I might rephrase to say that this is what allows you to do `[n,n'..]`.